### ISSUE TITLE
operator precedence error in parser

### ISSUE THREAD

**Issue by awcross1** [QISKit/qiskit-sdk-py#190]
The operator precedence appears to be incorrect while parsing expressions. I noticed this while testing PR #171.

```python
from qiskit import qasm
ast = qasm.Qasm(data="""OPENQASM 2.0;
include "qelib1.inc";
qreg qr[1];
creg cr[1];
u1(0.2*pi+0.3*pi) qr[0]; // precedence error in parser!
measure qr[0] -> cr[0];""").parse()
print(ast.qasm())
```

## Expected Behavior

Multiplication is expected to take precedence over addition.

## Current Behavior

The abridged output is
> qreg qr[1];
> creg cr[1];
> u1(((0.200000000000000*(3.141592653589793+0.300000000000000))*3.141592653589793)) qr[0];
> measure qr[0] -> cr[0];
on the current master branch (prior to accepting PR #171, which will replace pi by a symbol)

Here addition is taking precedence.

## Possible Solution

The problem is near line 955 in _qasmparser.py

## Steps to Reproduce (for bugs)

Run the example program above.

**Comment by adjs** [comment]
The grammar below (extracted from [here](https://arxiv.org/pdf/1707.03429.pdf)) is ambiguous and can lead to this kind of behavior because does not capture the notion of precedence.

exp -> 	exp + exp | exp - exp | exp * exp
		| exp / exp | - exp | exp ^ exp | (exp) | unaryop (exp)

A small example of a grammar that captures precedence from Sipser book.

expr -> expr + term | term
term -> term * factor | factor
factor -> (expr)| a

**Comment by adjs** [comment]
PR #193 should address this issue.

**Comment by diego-plan9** [comment]
Fixed by #193
